internal/controller/cephfscg/volumegroupsourcehandler.go

```diff
@@ -214,42 +214,48 @@ func (h *volumeGroupSourceHandler) CleanVolumeGroupSnapshot(
 	return nil
 }
 
-// RestoreVolumesFromVolumeGroupSnapshot restore VolumeGroupSnapshot to PVCs
+// RestoreVolumesFromVolumeGroupSnapshot restores VolumeGroupSnapshot to PVCs
+//
+//nolint:funlen,cyclop
 func (h *volumeGroupSourceHandler) RestoreVolumesFromVolumeGroupSnapshot(
 	ctx context.Context, owner metav1.Object,
 ) ([]RestoredPVC, error) {
 	logger := h.Logger.WithName("RestoreVolumesFromVolumeGroupSnapshot")
 	logger.Info("Get volume group snapshot")
 
-	volumeGroupSnapshot := &vgsv1alphfa1.VolumeGroupSnapshot{}
+	vgs := &vgsv1alphfa1.VolumeGroupSnapshot{}
 	if err := h.Client.Get(ctx,
 		types.NamespacedName{Name: h.VolumeGroupSnapshotName, Namespace: h.VolumeGroupSnapshotNamespace},
-		volumeGroupSnapshot); err != nil {
+		vgs); err != nil {
 		return nil, fmt.Errorf("failed to get volume group snapshot: %w", err)
 	}
 
-	if volumeGroupSnapshot.Status == nil || volumeGroupSnapshot.Status.ReadyToUse == nil ||
-		(volumeGroupSnapshot.Status.ReadyToUse != nil && !*volumeGroupSnapshot.Status.ReadyToUse) {
+	if vgs.Status == nil || vgs.Status.ReadyToUse == nil ||
+		(vgs.Status.ReadyToUse != nil && !*vgs.Status.ReadyToUse) {
 		return nil, fmt.Errorf("can't restore volume group snapshot: volume group snapshot is not ready to be used")
 	}
 
 	restoredPVCs := []RestoredPVC{}
 
-	for _, pvcVSRef := range volumeGroupSnapshot.Status.PVCVolumeSnapshotRefList {
+	for _, pvcVSRef := range vgs.Status.PVCVolumeSnapshotRefList {
 		logger.Info("Get PVCName from volume snapshot",
 			"PVCName", pvcVSRef.PersistentVolumeClaimRef.Name, "VolumeSnapshotName", pvcVSRef.VolumeSnapshotRef.Name)
 
 		pvc, err := util.GetPVC(ctx, h.Client,
-			types.NamespacedName{Name: pvcVSRef.PersistentVolumeClaimRef.Name, Namespace: volumeGroupSnapshot.Namespace})
+			types.NamespacedName{Name: pvcVSRef.PersistentVolumeClaimRef.Name, Namespace: vgs.Namespace})
 		if err != nil {
 			return nil, fmt.Errorf("failed to get PVC from VGS %s: %w",
-				volumeGroupSnapshot.Namespace+"/"+pvcVSRef.PersistentVolumeClaimRef.Name, err)
+				vgs.Namespace+"/"+pvcVSRef.PersistentVolumeClaimRef.Name, err)
 		}
 
-		restoreStorageClass, err := GetRestoreStorageClass(ctx, h.Client,
-			*pvc.Spec.StorageClassName, h.DefaultCephFSCSIDriverName)
+		storageClass, err := GetStorageClass(ctx, h.Client, pvc.Spec.StorageClassName)
 		if err != nil {
-			return nil, fmt.Errorf("failed to get Restore Storage Class from PVC %s: %w", pvc.Name+"/"+pvc.Namespace, err)
+			return nil, err
+		}
+
+		restoreAccessModes := []corev1.PersistentVolumeAccessMode{corev1.ReadOnlyMany}
+		if storageClass.Provisioner != h.DefaultCephFSCSIDriverName {
```

Comment:
(future) Upstream e2e the value for `DefaultCephFSCSIDriverName` should be `cephfs.csi.ceph.com` and potentially needs to be the default in the ramen config map.

Reply:
@ShyamsundarR can you elaborate more on this. The driver cephfs name is: `openshift-storage.cephfs.csi.ceph.com`

Reply:
@BenamarMk upstream the driver name is [cephfs.csi.ceph.com](https://github.com/ceph/ceph-csi/blob/853cecbee5217c65fb59ef6c1805c2aa953407a5/deploy/cephfs/kubernetes/csidriver.yaml#L12), so the comment pertains to changing ramen config map for this value upstream, OR do a substring match to this suffix for all cases.